packages/open-payments/src/client/index.ts

```diff
@@ -124,6 +127,11 @@ export const createUnauthenticatedClient = async (
       axiosInstance,
       openApi: resourceServerOpenApi,
       logger
+    }),
+    incomingPayment: createUnauthenticatedIncomingPaymentRoutes({
```

Comment:
it would be great if the public incoming payment was also possible to be fetched using the authenticated client (either through overloading `incomingPayment.get` somehow, or just adding `incomingPayment.getPublic()` (or something else maybe)?

Reply:
Something like this seems reasonable. I believe I can just remove the authorization header (via not passing in an `accessToken`) to get the public version.
```ts
export const createIncomingPaymentRoutes = (
  deps: RouteDeps
): IncomingPaymentRoutes => {

    ...
    return {
        ...
        
        getPublic: (args: ResourceRequestArgs) => {
          const { accessToken, ...argsWithoutAccessToken } = args
          return getPublicIncomingPayment(
            { axiosInstance, logger },
            argsWithoutAccessToken,
            getPublicIncomingPaymentOpenApiValidator
          )
        }
        
        ...
    }
    ...
    
```

Reply:
also considered the function overloading with an implementation like:

```ts
export const getIncomingPaymentWIP = async (
  deps: BaseDeps,
  args: ResourceRequestArgs | UnauthenticatedResourceRequestArgs,
  validateOpenApiResponse:
    | ResponseValidator<IncomingPaymentWithPaymentMethods>
    | ResponseValidator<PublicIncomingPayment>
) => {
  if ('accessToken' in args) {
    return getIncomingPayment(deps, args, validateOpenApiResponse as ResponseValidator<IncomingPaymentWithPaymentMethods>)
  } else {
    getPublicIncomingPayment(deps, args, validateOpenApiResponse as ResponseValidator<PublicIncomingPayment>)
  }
}
```

But then I think we're back to a union return type (an incoming payment OR public incoming payment).